planner: prepared plan cache support limit (#12307)
pingcap/docs: sql-prepared-plan-cache.md

- The query contains the `ignore_plan_cache` hint, such as `select /*+ ignore_plan_cache() */ * from t`.
- The query contains variables other than `?` (including system variables or user-defined variables), such as `select * from t where a>? and b>@x`.
- The query contains the functions that cannot be cached: `database()`, `current_user`, `current_role`, `user`, `connection_id`, `last_insert_id`, `row_count`, `version`, and `like`.

> - The query with `Count` being `?` after `Limit` and the `Count` value is greater than 10000.

Reviewer: ```suggestion
- The query with a variable as the `LIMIT` parameter (`LIMIT ?`) and the variable value is greater than 10000.
```
